grpc/src/main/java/io/stargate/grpc/service/ValuesHelper.java

```diff
@@ -189,72 +205,100 @@ public static TypeSpec convertType(ColumnType columnType) throws StatusException
+    try {
+      switch (columnType.rawType()) {
+        case List:
+          parameters = columnType.parameters();
+          if (parameters.size() != 1) {
+            throw Status.FAILED_PRECONDITION
+                .withDescription("Expected list type to have a parameterized type")
+                .asException();
+          }
+          builder.setList(
+              TypeSpec.List.newBuilder().setElement(convertType(parameters.get(0))).build());
+          break;
+        case Map:
+          parameters = columnType.parameters();
+          if (parameters.size() != 2) {
+            throw Status.FAILED_PRECONDITION
+                .withDescription("Expected map type to have key/value parameterized types")
+                .asException();
+          }
+          builder.setMap(
+              TypeSpec.Map.newBuilder()
+                  .setKey(convertType(parameters.get(0)))
+                  .setValue(convertType(parameters.get(1)))
+                  .build());
+          break;
+        case Set:
+          parameters = columnType.parameters();
+          if (parameters.size() != 1) {
+            throw Status.FAILED_PRECONDITION
+                .withDescription("Expected set type to have a parameterized type")
+                .asException();
+          }
+          builder.setSet(
+              TypeSpec.Set.newBuilder().setElement(convertType(parameters.get(0))).build());
+          break;
+        case Tuple:
+          parameters = columnType.parameters();
+          if (parameters.isEmpty()) {
+            throw Status.FAILED_PRECONDITION
+                .withDescription("Expected tuple type to have at least one parameterized type")
+                .asException();
+          }
+          TypeSpec.Tuple.Builder tupleBuilder = TypeSpec.Tuple.newBuilder();
+          for (ColumnType parameter : parameters) {
+            tupleBuilder.addElements(convertType(parameter));
+          }
+          builder.setTuple(tupleBuilder.build());
+          break;
+        case UDT:
+          UserDefinedType udt = (UserDefinedType) columnType;
+          if (udt.columns().isEmpty()) {
+            throw Status.FAILED_PRECONDITION
+                .withDescription("Expected user defined type to have at least one field")
+                .asException();
+          }
+          TypeSpec.Udt.Builder udtBuilder = TypeSpec.Udt.newBuilder();
+          for (Column column : udt.columns()) {
+            udtBuilder.putFields(column.name(), convertType(columnTypeNotNull(column)));
+          }
+          builder.setUdt(udtBuilder.build());
+          break;
+        default:
+          builder.setBasic(
+              Objects.requireNonNull(
+                  TypeSpec.Basic.forNumber(columnType.id()), "Unhandled parameterized type"));
+      }
+    } catch (StatusException statusException) {
+      LOG.error(getErrorMessage(statusException.getStatus(), statusException), statusException);
+      throw statusException;
```

Comment:
Usually logging AND rethrowing is an anti-pattern. Does log line here have enough information to be useful for us? Like contextual information on caller and so on?